File: backend/internal/model/monitor.go
package model

import "time"

type Monitor struct {
	ID              int64     `json:"id"`
	UserID          int64     `json:"user_id"`
	Name            string    `json:"name"`
	URL             string    `json:"url"`
	MonitorType     string    `json:"monitor_type"`
	IntervalSeconds int       `json:"interval_seconds"`
	IsActive        bool      `json:"is_active"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type CreateMonitorRequest struct {
	Name            string `json:"name" binding:"required,min=2,max=150"`
	URL             string `json:"url" binding:"required"`
	MonitorType     string `json:"monitor_type"`
	IntervalSeconds int    `json:"interval_seconds"`
	IsActive        *bool  `json:"is_active"`
}

type UpdateMonitorRequest struct {
	Name            *string `json:"name"`
	URL             *string `json:"url"`
	MonitorType     *string `json:"monitor_type"`
	IntervalSeconds *int    `json:"interval_seconds"`
	IsActive        *bool   `json:"is_active"`
}

type MonitorResult struct {
	URL          string `json:"url"`
	Status       string `json:"status"`
	HTTPStatus   int    `json:"http_status"`
	ResponseTime int64  `json:"response_time_ms"`
	CheckedAt    string `json:"checked_at"`
	Error        string `json:"error,omitempty"`
}

type MultipleCheckResponse struct {
	Total           int             `json:"total"`
	TotalDurationMs int64           `json:"total_duration_ms"`
	Results         []MonitorResult `json:"results"`
}

File: frontend/src/lib/types/monitor.ts
export type MonitorType = 'website' | 'api';

export interface Monitor {
	id: number;
	user_id: number;
	name: string;
	url: string;
	monitor_type: MonitorType;
	interval_seconds: number;
	is_active: boolean;
	created_at: string;
	updated_at: string;
}

export interface CreateMonitorPayload {
	name: string;
	url: string;
	monitor_type?: MonitorType;
	interval_seconds?: number;
	is_active?: boolean;
}

export interface UpdateMonitorPayload {
	name?: string;
	url?: string;
	monitor_type?: MonitorType;
	interval_seconds?: number;
	is_active?: boolean;
}

export interface MonitorListResponse {
	total: number;
	monitors: Monitor[];
}

export interface MonitorResponse {
	monitor: Monitor;
}

export interface MonitorMutationResponse {
	message: string;
	monitor: Monitor;
}

export interface DeleteMonitorResponse {
	message: string;
}
